lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1402,11 +1405,21 @@ def _create_node_runbook(
             azure_node_runbook.vhd = vhd
             azure_node_runbook.marketplace = None
             azure_node_runbook.shared_gallery = None
+            log.debug(
+                "there is no way to detect vhd generation, unless user provides it"
+                f" current vhd generation is {azure_node_runbook.hyperv_generation}"
+            )
         elif azure_node_runbook.shared_gallery:
             azure_node_runbook.marketplace = None
             azure_node_runbook.shared_gallery = self._parse_shared_gallery_image(
                 azure_node_runbook.shared_gallery
             )
+            azure_node_runbook.hyperv_generation = _get_gallery_image_generation(
+                self._get_detailed_sig(azure_node_runbook.shared_gallery)
+            )
+        elif azure_node_runbook.community_gallery_image:
+            azure_node_runbook.marketplace = None
```

Comment:
Is it finished? There should be logic to get vhd gen and parse image like shared gallery.

Reply:
updated